Respect user preferences during the build and fix the Clang build by
disabling PCH.
https://bugs.gentoo.org/921915
--- a/source/src/Makefile
+++ b/source/src/Makefile
@@ -6,7 +6,6 @@
 # found to have been caused by the g++ compiler in the past. This seems to have
 # been fixed now by relaxing the optimization that g++ does, so although we'll
 # continue using clang++ (just in case), you can use g++ if you prefer.
-CXX=clang++
 
 # call like 'DEBUGBUILD=1 make' to compile a debug version of AssaultCube.
 ifdef DEBUGBUILD
@@ -34,10 +33,8 @@ ifeq ($(ACDEBUG),yes)
 endif
 
 ifneq (,$(findstring clang,$(CXX)))
-	CXXFLAGS= -O3 -fomit-frame-pointer
 	override CXXFLAGS+= -Wall -fsigned-char
 else
-	CXXFLAGS= -O3 -fomit-frame-pointer
 	override CXXFLAGS+= -Wall -fsigned-char -ffast-math -rdynamic -Wno-format-zero-length
 	ifeq "$(shell expr 'gcc -dumpversion' \>= 8)" "1"
 		override CXXFLAGS+= -Wno-stringop-truncation
@@ -182,7 +179,6 @@ rendermodel.o: rendermodel.cpp
 	$(CXX) $(CXXFLAGS) -c -o $@ $(subst -standalone.o,.cpp,$@)
 
 $(CLIENT_OBJS): CXXFLAGS += $(CLIENT_INCLUDES)
-$(CLIENT_OBJS): $(CLIENT_PCH)
 $(SERVER_OBJS): CXXFLAGS += $(SERVER_INCLUDES)
 $(filter-out $(SERVER_OBJS),$(MASTER_OBJS)): CXXFLAGS += $(SERVER_INCLUDES)
 
